classes/migration/upgrade/v3_6_0/I10404_UpdateCategoryImageNameFields.php

```diff
@@ -0,0 +1,159 @@
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Enumerable;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use League\Flysystem\Filesystem;
+use League\Flysystem\Local\LocalFilesystemAdapter;
+use League\Flysystem\UnixVisibility\PortableVisibilityConverter;
+use PKP\config\Config;
+use PKP\core\Core;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+abstract class I10404_UpdateCategoryImageNameFields extends Migration
+{
+    private const FILE_MODE_MASK = 0666; // FileManager::FILE_MODE_MASK
+    private const DIRECTORY_MODE_MASK = 0777; // FileManager::DIRECTORY_MODE_MASK
+    public const CHUNK_SIZE = 1000;
+
+    public function up(): void
+    {
+        Schema::table('categories', function (Blueprint $table) {
+            $table->dropColumn('seq');
+        });
+
+        $contextIds = app()->get('context')->getIds();
+        $fileNamesToMove = [];
+
+        foreach ($contextIds as $contextId) {
+            DB::table('categories')
+                ->where('context_id', '=', $contextId)
+                ->orderBy('category_id')
+                ->chunk(self::CHUNK_SIZE, function (Collection $categories) use ($contextId, &$fileNamesToMove) {
+                    $imageRecordsToUpdate = [];
+
+                    foreach ($categories as $category) {
+                        $image = $category->image;
+                        if ($image) {
+                            $imageDecoded = json_decode($image, true);
+                            $fileNamesToMove[] = $imageDecoded['name'];
+                            $fileNamesToMove[] = $imageDecoded['thumbnailName'];
+                            // Swap 'name' and 'uploadName' fields
+                            [$imageDecoded['name'], $imageDecoded['uploadName']] = [$imageDecoded['uploadName'], $imageDecoded['name']];
+                            $imageRecordsToUpdate[$category->category_id] = json_encode($imageDecoded);
+                        }
+                    }
+
+                    $this->updateCategoriesImageNameFields($contextId, collect($imageRecordsToUpdate));
+                    // Move images
+                    $this->moveContextCategoryImagesToPublicFolder($contextId, collect($fileNamesToMove));
+                });
+        }
+    }
+
+    /**
+     *
+     * @param Enumerable $updates List of Category image data to update. Keyed by category ID.
+     */
+    private function updateCategoriesImageNameFields(int $contextId, Enumerable $updates): void
+    {
+        if ($updates->isEmpty()) {
+            return;
+        }
+
+        $caseStatement = 'UPDATE categories SET image = CASE category_id ';
+
+        foreach ($updates as $categoryId => $json) {
+            $caseStatement .= "WHEN {$categoryId} THEN ? ";
+        }
+
+        $caseStatement .= 'END WHERE category_id IN (' . implode(',', $updates->keys()->all()) . ') AND context_id = ?';
```

Comment:
It's possible to drop these `CASEs` with a string replace, I think I've mentioned it before, then you'll solve this issue with a single query, which will cover all contexts.

```sql
UPDATE categories SET image = REPLACE(REPLACE(REPLACE(image, '"name":', '"swapper":'), '"uploadName":', '"name":'), '"swapper":', '"uploadName":')
```